Failing redis-related builds

We can see this in the `react-gradle-cassandra-session-redis` builds

```
 AccountResourceIT > testRegisterDuplicateLogin() FAILED
    java.lang.IllegalStateException at DefaultCacheAwareContextLoaderDelegate.java:132
        Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:798
            Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:798
                Caused by: org.springframework.beans.factory.BeanCreationException at ConstructorResolver.java:656
                    Caused by: org.springframework.beans.BeanInstantiationException at SimpleInstantiationStrategy.java:185
                        Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:798
                            Caused by: org.springframework.beans.factory.BeanCreationException at ConstructorResolver.java:656
                                Caused by: org.springframework.beans.BeanInstantiationException at SimpleInstantiationStrategy.java:185
                                    Caused by: java.lang.IllegalArgumentException at RedisURI.java:42
```

I haven't found any related issue, but I may have not notice it

--- a/generators/server/templates/src/test/java/package/RedisTestContainerExtension.java.ejs
+++ b/generators/server/templates/src/test/java/package/RedisTestContainerExtension.java.ejs
@@ -21,21 +21,28 @@ package <%= packageName %>;
 
 import org.junit.jupiter.api.extension.BeforeAllCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
 import org.testcontainers.containers.GenericContainer;
 
 import java.util.concurrent.atomic.AtomicBoolean;
 
 public class RedisTestContainerExtension implements BeforeAllCallback {
+
     private static AtomicBoolean started = new AtomicBoolean(false);
+
+    private static GenericContainer redis;
+
+    @DynamicPropertySource
+    static void redisProperties(DynamicPropertyRegistry registry) {
+        registry.add("jhipster.cache.redis.server", () -> "redis://" + redis.getContainerIpAddress() + ":" + redis.getMappedPort(6379));
+    }
 
     @Override
     public void beforeAll(ExtensionContext extensionContext) throws Exception {
         if (!started.get()) {
-            GenericContainer redis =
-                new GenericContainer("<%= DOCKER_REDIS %>")
-                    .withExposedPorts(6379);
+            redis = new GenericContainer("<%= DOCKER_REDIS %>").withExposedPorts(6379);
             redis.start();
-            System.setProperty("redis.test.server", "redis://" + redis.getContainerIpAddress() + ":" + redis.getMappedPort(6379));
             started.set(true);
         }
     }

--- a/generators/server/templates/src/test/resources/config/application.yml.ejs
+++ b/generators/server/templates/src/test/resources/config/application.yml.ejs
@@ -235,11 +235,6 @@ jhipster:
         logs: # Reports metrics in the logs
             enabled: true
             report-frequency: 60 # in seconds
-<%_ if (cacheProvider === 'redis') { _%>
-    cache:
-        redis:
-            server: ${redis.test.server}
-<%_ } _%>
 <%_ if (messageBroker === 'kafka') { _%>
 kafka:
     bootstrap-servers: localhost:9092

